File: src/org/thoughtcrime/securesms/mms/PartUriParser.java
package org.thoughtcrime.securesms.mms;

import android.net.Uri;

import org.thoughtcrime.securesms.attachments.AttachmentId;

public class PartUriParser {

  private final Uri uri;

  public PartUriParser(Uri uri) {
    this.uri = uri;
  }

  public AttachmentId getPartId() {
    return new AttachmentId(getId(), getUniqueId());
  }

  private long getId() {
    return Long.parseLong(uri.getPathSegments().get(2));
  }

  private long getUniqueId() {
    return Long.parseLong(uri.getPathSegments().get(1));
  }

}

File: app/src/main/java/org/thoughtcrime/securesms/mms/PartUriParser.kt
package org.thoughtcrime.securesms.mms

import android.content.ContentUris
import android.net.Uri
import org.thoughtcrime.securesms.attachments.AttachmentId

/**
 * Parses the given [Uri] into either an [AttachmentId] or a [Long]
 */
class PartUriParser(private val uri: Uri) {
  val partId: AttachmentId
    get() = AttachmentId(id)

  private val id: Long
    get() = ContentUris.parseId(uri)
}
